### PR TITLE
Added limits test suite

### PR DESCRIPTION
**Pull request by gsramdasi**
Signed-off-by: Gaurav Ramdasi <10132364+gsramdasi@users.noreply.github.com>

## Description
The limits suite has two tests - cursor_txn_test and max_key_cnt_test. The latter takes over 16hrs to run since it checks the 200 billion key limit. To skip that test, use `--no-suite=long` in the `meson test` command line.

To run the whole suite: `meson test -C $BDIR --suite=limits`
To skip max_key_cnt_test: `meson test -C $BDIR --suite=limits --no-suite=long`

## Issue(s) Addressed
[NFSE-5397](https://jira.micron.com/jira/browse/NFSE-5397)

### CODE REVIEW

File: tests/functional/api/kvs_api_test.c [hunk 1]
```diff
@@ -183,31 +183,6 @@ MTF_DEFINE_UTEST(kvs_api_test, create_null_kvdb)
     ASSERT_EQ(EINVAL, hse_err_to_errno(err));
 }
 
-MTF_DEFINE_UTEST(kvs_api_test, create_max)
```
Comment (tristan957) [inline tests/functional/api/kvs_api_test.c:186]: Why was this test deleted?

Comment (gsramdasi) [reply tests/functional/api/kvs_api_test.c:186]: it's repeated in the limits test

Comment (tristan957) [reply tests/functional/api/kvs_api_test.c:186]: I don't think it makes sense to move it to a file called `cursor_txn_test.c`

Comment (gsramdasi) [reply tests/functional/api/kvs_api_test.c:186]: true, let me rename this file.

**Comment by tristan957** [comment]
Do you think it makes more sense to just fold what you wrote into the existing api tests?

**Review by alexttx — APPROVED** [review]

**Comment by tristan957** [comment]
I like that delineation.

> max kvses per kvdb

This is an API limit. We document it as `HSE_KVS_COUNT_MAX` in `include/hse/limits.h`.

**Comment by gsramdasi** [comment]
@tristan957 My understanding is that the distinction is that API tests should be testing all the variations of the arguments that could be passed to one call of the API, such as NULL handles, invalid characters in name etc. Not limits that are encountered as a result of a combination of multiple API calls - max cursors, txns or KVSes. These belong in the limits suite.

Max key and value length, as @alexttx said, could go anywhere. I kept it with limits so running that one suite ensures that all limits have been exercised.

File: tests/functional/limits/kvdb_limits_test.c [hunk 2]
```diff
@@ -0,0 +1,218 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```
Comment (tristan957) [inline tests/functional/limits/kvdb_limits_test.c:1]: Can you separate this out into individual files? I can at least see a `cursor_limits_test`, `transaction_limits_test`, `kvdb_limits_test`, and `kvs_limits_test`

Comment (gsramdasi) [reply tests/functional/limits/kvdb_limits_test.c:1]: Hmm, there are very few test cases here to split across multiple files. I think I'll leave it as is for now, and if we add more cases in the future, we can split them out.

**Review by nabeelmmd — APPROVED** [review]

**Pull request merged**